src/PooledArrays.jl

```diff
@@ -165,31 +186,120 @@ Base.size(pa::PooledArray) = size(pa.refs)
 Base.length(pa::PooledArray) = length(pa.refs)
 Base.lastindex(pa::PooledArray) = lastindex(pa.refs)
 
-Base.copy(pa::PooledArray) = PooledArray(RefArray(copy(pa.refs)), copy(pa.invpool))
-# TODO: Implement copy_to()
+Base.copy(pa::PooledArray) = PooledArray(pa)
+
+Base.copyto!(dest::PooledArray{T, R, N, RA},
+             src::PooledArray{T, R, N, RA}) where {T, R, N, RA} =
+    copyto!(dest, 1, src, 1, length(src))
+
+# TODO: handle case when dest is SubArray
+
+Base.copyto!(dest::PooledArray{T, R, N, RA},
+             src::SubArray{T, N, PooledArray{T, R, M, RA}}) where {T, R, N, M, RA} =
+    copyto!(dest, 1, src, 1, length(src))
+
+function Base.copy!(dest::PooledArray{T, R, N, RA},
+                    src::PooledArray{T, R, N, RA}) where {T, R, N, RA}
+    copy!(dest.refs, src.refs)
+    Threads.atomic_add!(src.refcount, 1)
+    dest.pool = src.pool
+    dest.invpool = src.invpool
+    dest.refcount = src.refcount
+    return dest
+end
+
+function Base.copy!(dest::PooledArray{T, R, N, RA},
+                    src::SubArray{T, N, PooledArray{T, R, M, RA}}) where {T, R, N, M, RA}
+    copy!(dest.refs, view(parent(src).refs, src.indices))
+    Threads.atomic_add!(src.refcount, 1)
+    dest.pool = src.pool
+    dest.invpool = src.invpool
+    dest.refcount = src.refcount
+    return dest
+end
+
+function Base.copyto!(dest::PooledArray{T, R, N, RA}, doffs::Union{Signed, Unsigned},
+                 src::PooledArray{T, R, N, RA}, soffs::Union{Signed, Unsigned},
+                 n::Union{Signed, Unsigned}) where {T, R, N, RA}
+    n == 0 && return dest
+    n > 0 || Base._throw_argerror()
+    if soffs < 1 || doffs < 1 || soffs + n - 1 > length(src) || doffs + n - 1 > length(dest)
+        throw(BoundsError())
+    end
+
+    if length(dest.pool) == 0
```

Comment:
```suggestion
    # if dest.pool is empty we can safely replace it as we are sure it holds
    # no information; having this path is useful because then we can efficiently
    # `copyto!` into a fresh `PooledArray` created using the `similar` function
    if length(dest.pool) == 0
```